overlord/fdestate/conflict.go

```diff
@@ -65,3 +71,78 @@ func checkFDEChangeConflict(st *state.State) error {
 	}
 	return nil
 }
+
+func dbxUpdateAffectedSnaps(t *state.Task) ([]string, error) {
+	// TODO:FDEM: check if we have sealed keys at all
+
+	// DBX updates cause a reseal, so any snaps which are either directly
+	// measured or their content is measured during the boot will count as
+	// affected
+
+	// XXX this effectively blocks updates of gadget, kernel & base until the
+	// change completes
+
+	return fdeRelevantSnaps(t.State())
+}
+
+func checkDBXChangeConflicts(st *state.State) error {
+	// TODO:FDEM: check if we have sealed keys at all
+
+	snaps, err := fdeRelevantSnaps(st)
+	if err != nil {
+		return err
+	}
+
+	if len(snaps) == 0 {
+		return nil
+	}
+
+	// make sure that there are no other DBX changes in progress
+	op, err := findFirstPendingExternalOperationByKind(st, "fde-efi-secureboot-db-update")
+	if err != nil {
+		return err
+	}
+
+	if op != nil {
+		return &snapstate.ChangeConflictError{
+			ChangeKind: "fde-efi-secureboot-db-update",
+			Message:    "cannot start a new DBX update when conflicting actions are in progress",
+		}
+	}
```

Comment:
@bboozzoo does it make sense to add a similar check to `checkFDEParametersChangeConflicts` given that a `fde-efi-secureboot-db-update` external operation would indeed cause a reseal and changing the FDE parameters?

Reply:
not sure, there should be a call to `checkFDEChangeConflict` somewhere in the caller anyway, which should pick up the change level conflict